### ISSUE TITLE
Additional details needed for keyCredential properties 

### ISSUE THREAD

**Issue by psignoret** [microsoftgraph/microsoft-graph-docs-contrib#9048]
Both the beta and v1.0 articles for the **keyCredential** type are missing details for the different properties, including (but not limited to):

* Allowed values for **type** and **usage**.
* The accepted combinations of **type** and **usage**, including those which are not supported.
* The expected content format of **key**, for the different allowed values of **type** and **usage**. (e.g. if **type** is `AsymmetricX509Cert` and **usage** is `Encrypt` or `Verify`, it should be the Base64 string of the DER-encoded X.509 certificate without the private key).

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: dd04739f-1ec0-9915-abb3-6a0cd4890e8a
* Version Independent ID: 90aa6e1d-6efa-5100-b51f-7766325086a6
* Content: [keyCredential resource type - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/keycredential?view=graph-rest-1.0)
* Content Source: [api-reference/v1.0/resources/keycredential.md](https://github.com/microsoftgraph/microsoft-graph-docs-contrib/blob/main/api-reference/v1.0/resources/keycredential.md)
* Product: **applications**
* Technology: **microsoft-graph**
* GitHub Login: @madansr7
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by FaithOmbongi** [comment]
@madansr7 - Can you assist? Thanks.

**Comment by madansr7** [comment]
@psignoret , the following page lists in the note section that the customer should only provide the public key, as well as type & usage in the Request body section. Is that what you are asking to add on the keyCredential resource page as well?
https://learn.microsoft.com/en-us/graph/api/application-addkey?view=graph-rest-1.0&tabs=http

**Comment by psignoret** [comment]
@madansr7 No, I'm saying keyCredential and all associated reference articles (including beta and v1.0, servicePrincipal, application, keyCredential, addKey, removeKey, etc.) need to be thoroughly and accurately documented. 

There are *at least* three possible values for **usage** (e.g. Verify, Encrypt, Sign, etc.), and *at least* three possible values for **type** (e.g. AsymmetricX509Cert, Symmetric, X509CertAndPassword, etc.), which gives *at least* 9 possible combinations. The use case of each combination should be stated, as well as the precise expected value for **key** in each case. For example, for the **usage**: *Verify*, **type**: *AsymmetricX509Cert* combination, saying "the public key value" is not enough, it needs to be something similar to "the Base64 string of the byte for the DER-encoded X.509 certificate without the private key". Unsupported combinations should be noted (and if any of those are accepted by the APIs, they should have associated internal bugs opened to safely block them moving forward).

The **beta** and **v1.0** documentation should be consistent and should accurately reflect supported and accepted values. (For example, [this](https://learn.microsoft.com/en-us/graph/api/resources/keycredential?view=graph-rest-beta) lists 9 possible usage values, [this](https://learn.microsoft.com/en-us/graph/api/application-addkey?view=graph-rest-1.0&tabs=http) suggests only two are accepted, but neither of them are actually correct.)
